app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala

```diff
@@ -68,32 +69,30 @@ trait BitcoinSRunner extends BitcoinSLogger {
         .resolve()
   }
 
-  lazy val configDataDir: Path = Paths.get(
-    baseConfig.getStringOrElse("bitcoin-s.datadir",
-                               AppConfig.DEFAULT_BITCOIN_S_DATADIR.toString))
-
-  lazy val datadirPath: Path = datadirPathOpt.getOrElse(configDataDir)
-
-  private lazy val networkStr: String =
-    baseConfig.getString("bitcoin-s.network")
-
-  private lazy val network: BitcoinNetwork =
-    BitcoinNetworks.fromString(networkStr)
-
-  lazy val datadir: Path = {
-    lazy val lastDirname = network match {
-      case MainNet  => "mainnet"
-      case TestNet3 => "testnet3"
-      case RegTest  => "regtest"
-      case SigNet   => "signet"
-    }
-    datadirPath.resolve(lastDirname)
-  }
+  /** Base directory for all bitcoin-s data */
```

Comment:
```scala
/** The datadir specified inside of the bitcoin-s.conf file under the key `bitcoin-s.datadir`. 
  * This will be used as the datadir if a --datadir option was not specified on the command line */ 
```

is more descriptive and useful in my opinion.

Reply:
This isn't true, it is the final datadir we will be using. Taking into account the `--datadir` option, `--conf` option, and the `bitcoin-s.conf` file

Reply:
ok well clearly I don't understand what you are doing, so let's add docs so it's easier to follow.